Add tertiary variant to `Button`

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #7912 

## Relevant technical choices

This PR adds a "tertiary" variant to the `Button` component and replaces all the usages of the `Link` component in the context of a secondary CTA.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

kuasha420 commented on assets/js/modules/tagmanager/components/setup/SetupForm.js:
```diff
@@ -224,10 +223,11 @@ export default function SetupForm( { finishSetup } ) {
 						{ /*
-						This "link" below will be rendered as a <button> but should not
-						trigger a form submit when clicked, hence the `type="button"`.
+						This <button> below but should not trigger a form submit
+						when clicked, hence the `type="button"`.
```
```suggestion
						This <button> below should not trigger a form submit
						when clicked, hence the `type="button"`.
```

kuasha420 commented on assets/sass/components/dashboard/wizard/_googlesitekit-wizard-step.scss:
```diff
@@ -92,10 +92,6 @@
-		@media (min-width: $bp-desktop) {
-			margin-left: $grid-gap-desktop;
-		}
+		margin-left: $grid-gap-phone / 2;
```
Not sure what's happening here. Can you help me understand this change?

nfmohit replied:
Sure.

This style is for the `Back` button in the legacy setup, here:

![image](https://github.com/google/site-kit-wp/assets/20284937/7a5337ab-4932-496e-8ac4-161880055928)

While using the link, it has a `margin-left` value of `24px` (`$grid-gap-desktop`).

However, when we go on to replace the link with our tertiary button (which already has a horizontal padding of `16px`), using the same `margin-left` value of `24px` creates a much larger white space between the actions. To prevent this, the above change aims to preserve the same `24px` spacing between the actions by applying a `margin-left` value of `8px` (`$grid-gap-phone / 2`), effectively preserving the same `24px` spacing (`16px` padding from button + `8px` `margin-left`). This is what we get:

![image](https://github.com/google/site-kit-wp/assets/20284937/baff3391-8db5-45ee-a169-744a16c1ab9c)

In mobile, `develop` applies a margin-left of `16px`. However, since our tertiary button already has a horizontal padding of `16px`, to create some space, we apply the same `8px` margin-left in all viewports.

Please let me know if this makes sense, thank you!

kuasha420 replied:
Thank you for the explanation. However, since this is effecting all viewports and `8px` is more or less a literal value here, I think we should use that instead of `$grid-gap-phone / 2` which is hard to reason about in this context. What do you think?

kuasha420 replied:
Thank you for the explanation. However, since this is effecting all viewports and `8px` is more or less a literal value here, I think we should use that instead of `$grid-gap-phone / 2` which is hard to reason about in this context. What do you think?

nfmohit replied:
Updated, please take a look, thank you!

kuasha420 commented on assets/sass/components/global/_googlesitekit-publisher-wins.scss:
```diff
@@ -157,11 +157,7 @@
-		@media (min-width: $bp-desktop) {
-			gap: $grid-gap-desktop;
-		}
+		gap: $grid-gap-phone / 2;
```
Same here as above, the change seems correct, but unrelated at first glance. Additional context would be helpful.

nfmohit replied:
This uses the same concept as explained in the response in the above comment. This is for the `BannerActions` component.

Before:
![image](https://github.com/google/site-kit-wp/assets/20284937/989de1ec-a44a-4627-8d11-e881b2dd0d03)

After:
![image](https://github.com/google/site-kit-wp/assets/20284937/da9eb002-ab5c-46a8-928d-eda65eec272d)

After (tertiary button hovered):
![image](https://github.com/google/site-kit-wp/assets/20284937/a961306d-6ff5-4c86-a633-0be126b0bf51)

kuasha420 commented on assets/sass/components/key-metrics/_googlesitekit-key-metrics-setup-cta.scss:
```diff
@@ -119,7 +119,7 @@
 			@media (min-width: $bp-nonMobile) {
 				display: flex;
-				gap: 20px;
+				gap: 4px;
```
Same here, why the gap so different now?

nfmohit replied:
This also uses the same concept as above.

The key metrics setup CTA widget applies a gap of `20px` between the two actions. Since the tertiary button already has a horizontal padding of `16px`, we change the `gap` to `4px` so that the visual white space remains the same.

Before:
![image](https://github.com/google/site-kit-wp/assets/20284937/41a1a576-d34f-470d-bb61-2984f4cc6f8b)

After:
![image](https://github.com/google/site-kit-wp/assets/20284937/5ccdf375-2478-4d0d-932d-116165a02901)

After (tertiary button hovered):
![image](https://github.com/google/site-kit-wp/assets/20284937/0dd28d91-b7fe-47e4-9a39-9409c03249d2)

kuasha420 commented on assets/js/googlesitekit/components-gm2/Button.stories.js:
```diff
@@ -61,6 +61,13 @@ DisabledButton.args = {
 	disabled: true,
 };
 
+export const TertiaryButton = Template.bind( {} );
```
It's surprising that we don't have VRT scenarios for button with states. ie. hover, active, focused etc. If possible, could you add them for this tertiary and other buttons? If it's too much trouble, we can file a follow up issue to explore this.

nfmohit replied:
That is a very good thought, thank you! However, unfortunately, I have almost reached towards the end of the estimate for this issue, so I have opened #8054 to address this. Please let me know if that sounds good, thanks!

kuasha420 replied:
Sounds good to me, thank you for filing the issue :tada:

kuasha420 reviewed (commented):
Thank you @nfmohit, fantastic work here. Just a couple of small things, and some clarification about a few CSS changes, but this is almost ready for merge. amazing work :tada:

kuasha420 reviewed (changes requested):
Ah, accidentally left a comment instead of request change. :sweat_smile:

nfmohit commented:
> Ah, accidentally left a comment instead of request change. 😅

No worries! Thank you for the detailed review. I have responded to the feedback. I should've actually added context around those changes in advance, my bad! Please let me know if this all looks good now.

Thanks!

kuasha420 reviewed (changes requested):
@nfmohit  Thank you for addressing the review feedback and explaining the style changes. Just one more suggestion, and this should be good to go.

nfmohit commented:
> @nfmohit Thank you for addressing the review feedback and explaining the style changes. Just one more suggestion, and this should be good to go.

Addressed, back to you for another pass, thank you!

kuasha420 reviewed (approved):
Thanks @nfmohit . LGTM :tada:
